### PR TITLE
internal upload obligations unit tests

### PR DESCRIPTION
**Pull request by NateHogan**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers.Tests.DataAccess/Shared/CommonDataAccessTests.cs [hunk 1]
```diff
@@ -75,5 +76,42 @@ public async Task GetCompetentAuthorityById_AuthorityShouldBeReturned(Guid id, s
...
+                var helper = new ModelHelper(database.Model);
+                var authorityId = Guid.Parse("A3C2D0DD-53A1-4F6A-99D0-1CCFC87611A8");
+
+                var dataAccess = new CommonDataAccess(database.WeeeContext);
+
+                await database.Model.SaveChangesAsync();
+
+                var result = await dataAccess.FetchCompetentAuthorityApprovedSchemes(Core.Shared.CompetentAuthority.England);
+
+                result.Schemes.Should().NotContain(x => x.SchemeStatus.Value == SchemeStatus.Rejected.Value);
```
Comment (meglenapnovoselska) [inline src/EA.Weee.RequestHandlers.Tests.DataAccess/Shared/CommonDataAccessTests.cs:113]: would you mind adding a test where  result.Schemes.Should().Contain(status.Value == Approved)

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerTests.cs [hunk 2]
```diff
@@ -212,5 +234,81 @@ public void PostSelectAuthority_WithInvalidModel_ReturnsHoldingViewWithViewModel
...
+            var result = controller.UploadObligations(model) as ViewResult;
+
+            //assert
+            bool triggerDownload = result.ViewBag.TriggerDownload;
+
+            triggerDownload.Should().BeTrue();
+        }
+
+        [Fact]
+        public async Task DownloadTemplateGet_CallsGetPcsObligationsCsv_Once()
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerTests.cs:301]: would you mind updating it to ShouldCallHandler or something

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.RequestHandlers.Tests.DataAccess/Shared/CommonDataAccessTests.cs (hunk 1 above)
Comment (NateHogan) [reply src/EA.Weee.RequestHandlers.Tests.DataAccess/Shared/CommonDataAccessTests.cs:113]: Added

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerTests.cs (hunk 2 above)
Comment (NateHogan) [reply src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerTests.cs]: Updated

**Pull request merged**